graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py

```diff
@@ -0,0 +1,1560 @@
+                # need to escape special chars to be used as separators
+                if label_config.separator in SPECIAL_CHARACTERS:
+                    separator = f"\\{label_config.separator}"
+                else:
+                    separator = label_config.separator
+                label_counts_df = (
+                    original_labels.select(label_col)
+                    .withColumn(label_col, F.explode(F.split(F.col(label_col), separator)))
+                    .groupBy(label_col)
+                    .count()
+                )
+            else:  # Single-label classification
+                label_counts_df = original_labels.groupBy(label_col).count()
+
+            # TODO: Verify correctness here
+            label_counts_dict: Counter = Counter()
+            for label_count_row in label_counts_df.collect():
+                row_dict = label_count_row.asDict()
+                # Label value to count
+                label_counts_dict[row_dict[label_col]] = row_dict["count"]
+
+            self.label_properties[node_or_edge_type][VALUE_COUNTS] = (
+                self.label_properties.get(VALUE_COUNTS, Counter()) + label_counts_dict
+            )
+        else:
+            raise RuntimeError(f"Invalid task type: {label_config.task_type}")
+
+    def _create_split_files_from_rates(
+        self,
+        input_df: DataFrame,
+        label_column: str,
+        split_rates: Optional[SplitRates],
+        output_path: str,
+        seed: Optional[int] = None,
+    ) -> Dict:
+        """
+        Given an input dataframe and a list of split rates creates the
+        split masks and writes them to S3 and returns the corresponding
+        metadata.json dict elements.
+
+        Parameters
+        ----------
+        input_df: DataFrame
+            Input dataframe for which we will create split masks.
+        label_column: str
+            The name of the label column. If provided, the values in the column
+            need to be not null for the data point to be included in one of the masks.
+            If an empty string, all rows in the input_df are included in one of train/val/test sets.
+        split_rates: Optional[SplitRates]
+            A SplitRates object indicating the train/val/test split rates.
+            If None, a default split rate of 0.9:0.05:0.05 is used.
+        output_path: str
+            The output path under which we write the masks.
+        seed: int
+            An optional random seed for reproducibility.
+
+        Returns
+        -------
+        The metadata dict elements for the train/test/val masks, to be added to the caller's
+        edge/node type metadata.
+        """
+        # If the user did not provide a split rate we use a default
+        split_metadata = {}
+        if split_rates is None:
+            split_rates = SplitRates(train_rate=0.9, val_rate=0.05, test_rate=0.05)
+        else:
+            if sum(split_rates.tolist()) != 1.0:
```

Comment:
I think the implementation would support it, since we are doing independent multinomial sampling for each row. If that's what GConstruct does I can change this check to be `<= 1.0`.

Reply:
OK seems like `multinomial` does not support that, as the probabilities need to sum to 1. We'd have to come up with an alternative efficient way to do distributed sampling, which I would consider outside the scope of this PR.

Reply:
OK seems like `multinomial` does not support that, as the [probabilities need to sum to 1](https://numpy.org/doc/stable/reference/random/generated/numpy.random.Generator.multinomial.html). We'd have to come up with an alternative efficient way to do distributed sampling, which I would consider outside the scope of this PR.